Use current time as start time for OMTG

FSchumacher opened the pull request:
## Description
Use current time as of actually starting the open model thread group 

## Motivation and Context
When we use the test start time via JMeterContextService#getTestStartTime, we might create a storm of test samples at the beginning, when a setup thread group made us wait.

Related to #6352 

## How Has This Been Tested?
Used the test plan attached and looked at the req/s values. They should be in the range of 1 req/s.
[omtg-and-setup-group.jmx.zip](https://github.com/user-attachments/files/17116744/omtg-and-setup-group.jmx.zip)

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Delete as appropriate -->
- Bug fix (non-breaking change which fixes an issue)
- *Might be a* Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [code style][style-guide] of this project.
- [ ] I have updated the documentation accordingly.

[style-guide]: https://wiki.apache.org/jmeter/CodeStyleGuidelines

FSchumacher committed:
Use current time as start time for OMTG

When we use the test start time via JMeterContextService#getTestStartTime,
we might create a storm of test samples at the beginning, when a setup thread group
made us wait.

vlsi commented on src/core/src/main/kotlin/org/apache/jmeter/threads/openmodel/OpenModelThreadGroup.kt:
```diff
@@ -204,7 +204,7 @@ public class OpenModelThreadGroup :
             val rnd = if (seed == 0L) Random() else Random(seed)
             val gen = ThreadScheduleProcessGenerator(rnd, parsedSchedule)
-            val testStartTime = JMeterContextService.getTestStartTime()
+            val testStartTime = System.currentTimeMillis()
```
Do we have something like "setup completed timestamp" or "main test started (excluding setup)"?

FSchumacher replied:
I am not aware of such a thing. 

But do you think this is needed? ThreadGroup#start gets called, when the thread group should start, so this seems to enough (for now?).

vlsi replied:
`JMeterContextService.getTestStartTime` is used in `ConstantThroughputTimer`, `PreciseThroughputTimer`.
I expect the classes will have the same issue regarding long setup thread groups.
At the same time, all of those might appear under setup thread groups as well. WDYT of using something like "thread group start time"?

Of course, for `OMTG` it is a bit easier as `start` corresponds to the thread group start, however, the timers should probably refer to the start time of the corresponding thread groups.

FSchumacher replied:
Right, if they use the same timestamp to calculate the beginning of the thread group, they are misled, too. 

Do the timers have access to their respective thread groups?

vlsi replied:
There's `JMeterContextService.getContext().getThreadGroup()`.
However, `AbstractThreadGroup` does not track its start time :-/

vlsi replied:
It sounds like a good plan.

`ConstantThroughputTimer` has "shared" mode which means it somehow operates across the thread groups, so I am not sure which "start" timestamp it should use then.

FSchumacher replied:
Yes, there has to be done more logic in the `ConstantThrouputTimer`. The delay has to be calculated depending on the mode. So I left it out now, but it could be added later.

FSchumacher committed:
Add startTime to AbstractThreadGroup and use it in OMTG and PreciseThroughputTimer

A thread group might start later than the test. When we calculate delays on the start time of
the test, we might get a stampede of samples run at the start of the thread group.

vlsi commented on src/core/src/main/java/org/apache/jmeter/threads/AbstractThreadGroup.java:
```diff
@@ -86,6 +87,8 @@ public abstract class AbstractThreadGroup extends AbstractTestElement
 
     private final AtomicInteger numberOfThreads = new AtomicInteger(0); // Number of active threads in this group
 
+    private final AtomicLong startTime = new AtomicLong(0);
```
Can we use a regular `long` here? If we need to synchronize get/set calls, then we should probably document it

FSchumacher replied:
I thought more about read semantics in multithread environment. I thought of using a volatile long instead. Would you like that more?

FSchumacher replied:
That may be true for the thread group, but is true for the timers?

vlsi replied:
I guess the sequence is:
1) Thread group initializes
2) Start time is set
3) Thread group starts
4) Timers operate

The timers should not modify "start time", so there's no need to synchronize the access.

FSchumacher committed:
Use simple long, as we are initialized before the reading threads are started

FSchumacher committed:
Add since annotation to javadoc for new feature

vlsi reviewed (approved):
LGTM

FSchumacher commented:
Has been merged into master
